### PR TITLE
Convert loader to ESM

### PR DESCRIPTION
**Pull request by dcodeIO**
This PR converts the loader to ESM format so it is more easily usable both on the web and under node.js. I stumbled over this while working on the new website and its editor, where I need a little hack to make the loader work. Of course that'd be easily fixable by appending an AMD/CommonJS shim, which I considered first, but with https://github.com/AssemblyScript/assemblyscript/issues/1306 being a thing now (see for context) I figured that I can as well initiate the switch to ESM at the loader.

Thoughts? :)

- [x] I've read the contributing guidelines

### CODE REVIEW

**Comment by dcodeIO** [comment]
Well, seems TypeScript isn't happy with that in the bootstrap test. Also quickly looked up how to make a hybrid module, but that requires either transpiling, which is silly, or keeping the base code in CommonJS format, which isn't helpful in the browser.

**Comment by dcodeIO** [comment]
[insert random swearwords / honest opinion here]

Looks like the only thing that'll actually work in all circumstances is to transpile, but pulling in potentially heavy dependencies like babel just for that feels wrong as well :(

**Comment by jtenner** [comment]
Why not go the whole nine yards and convert the loader to typescript? :)

**Comment by dcodeIO** [comment]
That's indeed another option given the circumstances. What I'm trying to achieve currently is that we have future proof sources in ESM, and would make the code base directly use the sources so making changes is immediately in sync without a build step, and contributors don't get confused by multiple files that look like sources. Then, if we absolutely need a build step, automate it on publish.

**Comment by dcodeIO** [comment]
Latest commit now gives this a shot using [esm2umd](https://github.com/dcodeIO/esm2umd) as a prepublish step, making the loader default to ESM with UMD as a fallback. `ts-node` doesn't like this, unfortunately, perhaps because we are stuck on v6, so I converted the bootstrap test to ESM as well.

File: .eslintrc.js [hunk 1]
```diff
@@ -87,13 +87,18 @@ module.exports = {
...
       },
       files: [
-        "**/*.js"
+        "**/*.js",
+        "**/*.mjs",
+        "**/*.cjs"
       ],
       rules: {
         // Node's support for ESM is still not great, but this rule is likely
         // to become activated once compatibility doesn't suck anymore.
```
Comment (jtenner) [inline .eslintrc.js:96]: Maybe it would be a good idea to remove this language from our source?

**Comment by dcodeIO** [comment]
Superseded by https://github.com/AssemblyScript/assemblyscript/pull/1513

**Pull request closed without merging**